lib/KMultiSelect.vue

```diff
@@ -0,0 +1,1084 @@
+            top: '50%',
+            transform: 'translateY(-50%)',
+            fontSize: '16px',
+            zIndex: 1,
+          }"
+          aria-hidden="true"
+        />
+
+        <input
+          ref="comboboxInput"
+          v-model.trim="searchText"
+          type="text"
+          role="combobox"
+          :class="['combobox-input', $computedClass(inputStyles)]"
+          :style="{
+            width: '100%',
+            height: '40px',
+            padding: autocomplete ? '0 40px 0 40px' : '0 40px 0 12px',
+            border: 'none',
+            outline: 'none',
+            fontSize: '14px',
+            color: $themeTokens.text,
+          }"
+          :placeholder="placeholder"
+          :aria-label="comboboxAriaLabel"
+          :aria-expanded="isDropdownOpen.toString()"
+          :aria-controls="listboxId"
+          :aria-activedescendant="getActiveDescendant()"
+          :aria-describedby="ariaDescribedById"
+          :aria-autocomplete="autocomplete ? 'list' : 'none'"
+          :aria-required="required ? 'true' : 'false'"
+          :aria-invalid="invalid ? 'true' : 'false'"
+          :disabled="disabled"
+          :readonly="!autocomplete"
+          @input="handleInput"
+          @keydown="handleComboboxKeydown"
+          @focus="handleInputFocus"
+          @blur="handleInputBlur"
+          @click="handleInputClick"
+        >
+
+        <KIconButton
+          v-if="searchText && autocomplete"
+          size="small"
+          icon="clear"
+          class="clear-search-button"
+          :style="{
+            position: 'absolute',
+            right: '32px',
+            top: '50%',
+            transform: 'translateY(-50%)',
+          }"
+          :ariaLabel="clearSearchMessage"
+          @click="clearSearch"
+        />
+
+        <KIconButton
+          size="small"
+          :icon="isDropdownOpen ? 'chevronUp' : 'chevronDown'"
+          class="dropdown-toggle"
+          :style="{
+            position: 'absolute',
+            right: '8px',
+            top: '50%',
+            transform: 'translateY(-50%)',
+          }"
+          :ariaLabel="getToggleDropdownLabel()"
+          :aria-expanded="isDropdownOpen.toString()"
+          @click="toggleDropdown"
+          @mousedown.prevent
+        />
+      </div>
+
+      <!-- Dropdown List -->
+      <div
+        v-show="isDropdownOpen"
+        ref="dropdownContainer"
+        class="dropdown-container"
+        :style="{
+          position: 'absolute',
+          top: '100%',
+          left: '0',
+          right: '0',
+          zIndex: '1000',
+          backgroundColor: $themeTokens.surface,
+          border: `1px solid ${$themeTokens.fineLine}`,
+          borderTop: 'none',
+          borderRadius: '0 0 4px 4px',
+          maxHeight: '400px',
+          overflowY: 'auto',
+        }"
+        role="region"
+        :aria-live="autocomplete ? 'polite' : 'off'"
+        @mousedown.prevent
+      >
+        <ul
```

Comment:
When I open the dropdown, I hear all options announced one by one. That's not desired - imagine a screen reader reading 200 options aloud :) I just checked Alex's implementation that Radina (a11y lead) approved to be sure, and it doesn't do that either. If you'd like to hear the expected screen reader output for the list part, in Kolibri repository you can `git checkout abc1f0cc6acc1b45fcd68769728bf95a646d2e80` and then navigate to `http://localhost:8000/en/facility/#/classes` where there is a live example.